File: src/main/java/no/nav/permitteringsskjemaapi/Yrkeskategori.java
package no.nav.permitteringsskjemaapi;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import java.util.UUID;

// Lombok
@Data
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
// JPA
@Entity
public class Yrkeskategori {
    @Id
    @JsonIgnore
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "permitteringsskjema_id")
    @JsonIgnore
    @ToString.Exclude
    private Permitteringsskjema permitteringsskjema;

    private Integer konseptId;
    private String styrk08;
    private String label;
    private Integer antall;
}

File: src/main/kotlin/no/nav/permitteringsskjemaapi/permittering/Yrkeskategori.kt
package no.nav.permitteringsskjemaapi.permittering

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import java.util.*

@Entity
data class Yrkeskategori(
    @JsonIgnore
    @field:Id
    var id: UUID? = null,
    @JsonIgnore
    @field:ManyToOne
    @field:JoinColumn(name = "permitteringsskjema_id")
    var permitteringsskjema: Permitteringsskjema? = null,
    var konseptId: Int? = null,
    var styrk08: String? = null,
    var label: String? = null,
    var antall: Int? = null,
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Yrkeskategori

        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        return id?.hashCode() ?: 0
    }
}